src/newcharacter.cpp

```diff
@@ -2293,8 +2296,16 @@ tab_direction set_description(WINDOW *w, player *u, const bool allow_reroll, poi
         const std::string action = ctxt.handle_input();
 
         if (action == "NEXT_TAB") {
-            if( OPTIONS["POINT_DISTRIBUTION"] != "freeform" && !points.is_valid() ) {
-                popup(_("Too many points allocated, change some features and try again."));
+            if (OPTIONS["POINT_DISTRIBUTION"] != "freeform" && !points.is_valid() ) {
+                if( points.skill_points_left() < 0 ) {
+                        popup(_("Too many points allocated, change some features and try again."));
+                } else if( points.trait_points_left() < 0 ) {
+                        popup(_("Too many trait points allocated, change some traits or lower some stats and try again."));
+                } else if( points.stat_points_left() < 0 ) {
+                        popup(_("Too many stat points allocated, lower some stats and try again."));
+                } else {
+                        popup(_("Too many points allocated, change some features and try again."));
+                }
```

Comment:
I see you've merged the PR anyway; should I still investigate and see if the order could be improved, or has something changed?